src/server/pa_rot.rs

```diff
@@ -257,6 +267,34 @@ mod test {
         Ok(resp_val)
     }
 
+    fn simulate_request<'a, C: protocol::Command<'a>, A: Arena>(
+        scratch_space: &'a mut [u8],
+        arena: &'a mut A,
+        server: &mut PaRot<fake::Identity, fake::Reset, ring::rsa::Builder>,
+        request: C::Req,
+    ) -> Result<C::Resp, Error> {
+        simulate_request_to_response::<C, A, C::Resp>(
+            scratch_space,
+            arena,
+            server,
+            request,
+        )
+    }
+
+    fn simulate_request_to_error<'a, C: protocol::Command<'a>, A: Arena>(
+        scratch_space: &'a mut [u8],
+        arena: &'a mut A,
+        server: &mut PaRot<fake::Identity, fake::Reset, ring::rsa::Builder>,
+        request: C::Req,
+    ) -> Result<protocol::Error, Error> {
+        simulate_request_to_response::<C, A, protocol::Error>(
+            scratch_space,
+            arena,
+            server,
+            request,
+        )
+    }
```

Comment:
Perhaps instead it makes sense for `simulate_request` to return `Result<Result<Resp, protocol::Error>, Error>`, which you can interrogate via the returned header's command type.

Reply:
Good idea, thank you! Done.